api/rest/v2/observation.md

```diff
@@ -10,95 +10,193 @@ permalink: /api/rest/v2/observation
-entity and variable at a particular date. For example, “population of USA in
-2020”, “GDP of California in 2010”, “predicted temperature of New York in 2050”,
-and so on.
+The Observation API fetches statistical observations. An observation is associated with an
+entity and variable at a particular date: for example, "population of USA in
+2020", "GDP of California in 2010", and so on.
 
-When querying observations, you need to provide variables, entities, and date.
-Variables are specified as a list in the form of
+## Request
 
-```json
+<div class="api-tab">
+  <button id="get-button" class="api-tablink" onclick="openTab(event, 'GET-request')">
+    GET request
+  </button>
+  <button id="post-button" class="api-tablink" onclick="openTab(event, 'POST-request')">
+    POST request
+  </button>
+</div>
+
+<div id="GET-request" class="api-tabcontent api-signature">
+https://api.datacommons.org/v2/observation?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI&date=<var>DATE_EXPRESSION</var>&variable.dcids=<var>DCID_LIST</var>&entity.dcids|expression=<var>DCID_LIST_OR_RELATION_EXPRESSION</var>&select=variable&select=entity[&select=value][&select=date]
+</div>
+
+<div id="POST-request" class="api-tabcontent api-signature">
+URL:
+https://api.datacommons.org/v2/observation
+
+Header:
+X-API-Key: AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI
+
+JSON data:
 {
-  "dcids": ["<variable_dcid_1>", "<variable_dcid_2>"]
+  "date": "<var>DATE_EXPRESSION</var>",
+  "variable.dcids": [
+      "<var>VARIABLE_DCID_1</var>",
+      "<var>VARIABLE_DCID_2</var>",
+      ...
+    ],
+  "entity.dcids": [
+      "<var>ENTITY_DCID_1</var>",
+      "<var>ENTITY_DCID_2</var>",
+      ...
+    ],
+  "entity.expression": <var>ENTITY_EXPRESSION</var>,
+  "select": "variable",
+  "select": "entity",
+  ...
 }
-```
+</div>
+
+<script src="/assets/js/syntax_highlighting.js"></script>
+<script src="/assets/js/api-doc-tabs.js"></script>
+
+### Query parameters
+
+| Name                                                  | Type   |  Description                                                    |
+|-------------------------------------------------------|--------|-----------------------------------------------------------------|
+| key <br /> <required-tag>Required</required-tag>      | string | Your API key. See the [page on authentication](/api/rest/v2/getting_started.html#authentication) for a demo key, as well as instructions on how to get your own key. |
+| date <br /> <required-tag>Required</required-tag>     | string | See [below](#date-string) for allowable values. |
+| variable.dcids <br /> <required-tag>Required</required-tag>| list of strings | List of [DCIDs](/glossary.html#dcid) for the statistical variable to be queried. |
+| entity.dcids                                          | list of strings | Comma-separated list of [DCIDs](/glossary.html#dcid) of entities to query. At least one of `entity.dcids` or `entity.expression` is required. |
```

Comment:
instead of `At least one of entity.dcids or entity.expression is required`, should it be `One of entity.dcids or entity.expression is required`? because I don't think we can have both, can we?

Reply:
Yes, you can, as described later.

Reply:
oh interesting, where is this described?

Reply:
Whoops, I mixed this up with the select stuff. D'oh! It seemed like it should work (like I want the person count for Pittsburgh and all cities contained in California), but I just tried it and you're right, it doesn't actually work. It's not rejected as an invalid query, but entity.dcids seems to always take precedence (I tested using 2 orders). Do you think this is intended behavior or a bug?

Reply:
I've always assumed you can only do one or the other, but I'm not sure if that was the intention ..

Reply:
OK, so for now, I've corrected the text with the current behavior. When Bo gets back I'll try to remember to ask him if this is intended behavior. (Ideally it seems to me that you should be able to combine them, as you can have multiple entity.dcids)

Reply:
this sounds good to me